clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -1780,6 +1780,20 @@ LValue CIRGenFunction::emitLoadOfReferenceLValue(Address refAddr,
                         pointeeBaseInfo);
 }
 
+void CIRGenFunction::emitTrap(mlir::Location loc, bool createNewBlock) {
```

Comment:
Should the existing trap and unreachable emission in `emitImplicitReturn` be updated to use these functions? At least in the case of unreachable, the current code will miss the future sanitizer handling.

Reply:
I'd rather not do this change in this PR. The biggest reason is, in classic CodeGen, the sanitize calls for the unreachable emission in `emitImplicitReturn` are different from the sanitizer calls emitted in `emitUnreachable`, see https://godbolt.org/z/c1v9bnc8d:

- An explicit call to `__builtin_unreachable` emits sanitizer call to `__ubsan_handle_builtin_unreachable`, which is what will be emitted in `emitUnreachable`;
- A missing return in `emitImplicitReturn` emits sanitizer call to `__ubsan_handle_missing_return`.

`emitUnreachable` and `emitTrap` also (optionally) resets the insertion point of the IR builder, which is not that helpful for code in `emitImplicitReturn`.

Reply:
I added a missing feature call for sanitizers in `emitImplicitReturn` to track this.